File: app/src/main/java/com/wafflestudio/snutt2/model/Tag.java
package com.wafflestudio.snutt2.model;

/**
 * Created by makesource on 2016. 2. 21..
 */
public class Tag {

    private String name;
    private TagType tagType;

    public Tag(String name,TagType tagType) {
        this.name = name;
        this.tagType = tagType;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setTagType(TagType tagType) {
        this.tagType = tagType;
    }

    public TagType getTagType() {
        return tagType;
    }

}

File: core/model/src/main/java/com/wafflestudio/snutt2/core/model/data/Tag.kt
package com.wafflestudio.snutt2.core.model.data

data class Tag(
    val type: Type,
    val name: String,
) {
    companion object {
        val TIME_EMPTY: Tag = Tag(Type.TIME, "빈 시간대로 검색")
        val TIME_SELECT: Tag = Tag(Type.TIME, "시간대 직접 선택")
        val ETC_ENG: Tag = Tag(Type.ETC, "영어진행 강의")
        val ETC_MILITARY: Tag = Tag(Type.ETC, "군휴학 원격수업")
    }

    enum class Type {
        CLASSIFICATION,
        DEPARTMENT,
        ACADEMIC_YEAR,
        CREDIT,
        TIME,
        CATEGORY,
        ETC,
    }
}
